Version 2

anilanar opened the pull request:
Splits api into two:

1) `interface Api { mount(opts): ActionResult<string, Observable<Messenger>>, __unstableConsumeToken(...) }`
2) `interface Messenger { ... everything else from v1 except mount, unmount, __unstableConsumeToken ... }`

The idea is `Observable<Messenger>` to emit everytime UMM restarts (due to widget router resolving to new config, or UMM restarting for other reasons such as to recover from errors) and that the messenger is unmounted when the observable is unsubscribed from.

This separation also makes the behavior of what happens when api methods are called while UMM is unmounted much clearer as those Api methods are now very well scoped to while UMM is mounted.

Currently, codesandboxes (see README) have hardcoded playground9 environment as the base url. Will need to change that after UMM PR is merged and before this PR is merged.

anilanar committed:
feat: add new api version v2

Version 2 introduces enchancements that enable website routers for single page applications.

anilanar committed:
Publish

 - @userlike/messenger@1.3.0-alpha.0
 - @userlike/messenger-internal@3.2.0-alpha.0

anilanar committed:
feat: improve v2 type names

anilanar committed:
Publish

 - @userlike/messenger@1.3.0-alpha.1
 - @userlike/messenger-internal@3.2.0-alpha.1

anilanar committed:
fix: fix return type of v2 mount

anilanar committed:
Publish

 - @userlike/messenger@1.3.0-alpha.2
 - @userlike/messenger-internal@3.2.0-alpha.2

anilanar committed:
docs: update README for v2

buscape commented on packages/messenger-internal/src/v2/types.ts:
```diff
@@ -0,0 +1,34 @@
+  ): Promise<ActionResult<string, Credentials>>;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
```
nit: already handled in eslint config
```suggestion
```

buscape commented on packages/messenger-internal/src/shared/types.ts:
```diff
@@ -0,0 +1,135 @@
+  id: string;
+}
+
+export enum EnableRegistration {
```
nit: this enum is never used and can be `@deprecated`

buscape reviewed (approved):
lgtm

anilanar committed:
chore: clean up

anilanar committed:
refactor: rename ActionResult.ts to Result.ts

anilanar committed:
feat: rename ActionResult to Result and swap generic args

anilanar committed:
feat: extend mount api to return reason when no messenger mounts

anilanar committed:
Publish

 - @userlike/messenger@1.3.0-alpha.3
 - @userlike/messenger-internal@3.2.0-alpha.3

anilanar committed:
docs: add migration guide
